### PR TITLE
Added automated builds for Linux, OSX and Windows.

### PR DESCRIPTION
**Pull request by ghost**
### Summary
Automated builds for the CMake+QT projects in Debug and Release for all the three desktop platforms. Linux (gcc only) and OSX (clang only) use [Travis CI](https://travis-ci.org/), Windows uses [AppVeyor](https://www.appveyor.com/). Click the badges below to see the output, these have been embedded in the README.

 [Build Status](https://travis-ci.org/louis-mclaughlin/GamePlay)  [Build status](https://ci.appveyor.com/project/louis-mclaughlin/gameplay)

### Action required
@seanpaultaylor , you'll need to sign up for and enable Travis CI and AppVeyor on the gameplay3d/GamePlay and gameplay3d/GamePlay-deps repos. Once you've done that, you'll then need to update the README's in both repos to point at the build statuses for gameplay3d account rather than my account.

### Notes
- Fixes for linker errors when building with Ubuntu versions > 16.04
- Fixes for linker errors on OSX
- Fixes for compilation errors in Linux and OSX builds.
- Building the editor on Linux resulted in a lot of compilation errors due to an X11 header included in by SDL.  The header defines a lot of constants in the global namespace, this causes conflicts with some of the Qt header declarations. There are comments in the Qt headers about this conflict, I followed their suggestion of including Qt headers before X11 to resolve this, see "gameplay.h" in the editor project.

### CODE REVIEW

**Comment by seanpaultaylor** [comment]
I will signup for Travis CL and AppVeyer shortly after the merge

File: gameplay-editor/src/gameplay.h
Comment (seanpaultaylor) [inline gameplay-editor/src/gameplay.h]: We need to find another place to put this like Platform.h
I dont want SDL in gameplay.h

File: gameplay-editor/src/gameplay.h
Comment (seanpaultaylor) [inline gameplay-editor/src/gameplay.h]: This should be done via path resolution and not redirect the header file with the same header file in a different path

File: gameplay-editor/src/gameplay.h
Comment (seanpaultaylor) [inline gameplay-editor/src/gameplay.h]: Also I don't want to have any Qt in the gameplay.h
Only the gameplay-editor should require Qt.

**Review by seanpaultaylor — CHANGES_REQUESTED** [review]
Need to make some minor changes to address the gameplay.h changes then this change looks good.

**Comment by ghost** [comment]
Given these constraints I could only see two solutions:
* Remove the header, fix all the existing editor includes of gameplay.h so that they include the Qt headers first. Not great, would add several qt includes per editor header and anyone writing new code would have to be aware of this gotcha in order not to break the Linux build
* Remove the header, un-define all the known conflicting defines after the include of SDL.h in Platform.h. Any new code that conflicts with an existing define would need to add another un-define to Platform.h, probably won't occur that often.

I've gone for the second option, open to suggestions.

**Comment by seanpaultaylor** [comment]
Let me try to get the code so it does not require Platform for editor builds. We can add a GP_EDITOR definition and use opaque pointers instead of SDLWindow*

**Comment by xahon** [comment]
Offtop

This is the only place I can communicate to maintainers. Why doesn't it have issues tab?

**Comment by seanpaultaylor** [comment]
The issues was disabled when v3.0.0 was no longer taking future updates. 
We are now working on getting the basics working for v4.0.0. After the v4.0.0 alpha which is still a while out, we will re-enable the issue to track bugs and features. If you want to participate in on gameplay3d-dev let me know. Our dev team uses Discord for this communication.

**Pull request merged**
